packages/edit-site/src/components/layout/router.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { useIsSiteEditorLoading } from './hooks';
+import Editor from '../editor';
+import DataviewsPatterns from '../page-patterns/dataviews-patterns';
+import PagePages from '../page-pages';
+import PagePatterns from '../page-patterns';
+import PageTemplateParts from '../page-template-parts';
+import PageTemplates from '../page-templates';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+export default function useLayoutAreas() {
+	const isSiteEditorLoading = useIsSiteEditorLoading();
+	const { params } = useLocation();
+	const { postType, postId, path, type, isCustom } = params ?? {};
+
+	// Regular page
+	if ( path === '/page' ) {
+		const isListLayout =
+			isCustom !== 'true' && ( ! type || type === 'list' );
+		return {
+			areas: {
+				content: window.__experimentalAdminViews ? (
+					<PagePages />
+				) : undefined,
+				preview: ( isListLayout ||
+					! window.__experimentalAdminViews ) && (
+					<Editor isLoading={ isSiteEditorLoading } />
+				),
+			},
+			widths: {
+				content:
+					window.__experimentalAdminViews && isListLayout
+						? 380
+						: undefined,
+			},
+		};
+	}
+
+	// Regular other post types
+	if ( postType && postId ) {
+		return {
+			areas: {
+				preview: <Editor isLoading={ isSiteEditorLoading } />,
+			},
+		};
+	}
+
+	// Templates
+	if (
+		path === '/wp_template/all' ||
+		( path === '/wp_template' && window?.__experimentalAdminViews )
+	) {
+		const isListLayout =
+			isCustom !== 'true' && ( ! type || type === 'list' );
+		return {
+			areas: {
+				content: <PageTemplates />,
+				preview: isListLayout && (
+					<Editor isLoading={ isSiteEditorLoading } />
+				),
+			},
+			widths: {
```

Comment:
The concept of widths in a router 'API' is weird. I'm wondering if we could apply this in the `content` prop and have some css rules in the layout containers like `width: max-content` or something(I haven't tried anything to see if it's possible with css).

Reply:
It's possible but the reason I kept it here for now is that I thought that maybe the "resizing" that is implemented in layout could use that width at some point.